tools/dev/spec_zero_update_versions.py

```diff
@@ -0,0 +1,133 @@
+"""Update dependency version specifiers to comply with SPEC0."""
```

Comment:
Can you add some lines here about which deps this updates? Just required? `full*` or `hdf5` install variants?

Please also add/recreate some verbiage about the update policy we settled on in this thread, and add it here. That way it's documented at the top exactly what this script will try to do. It's not 100% the same as SPEC0. I'm fine with leaving that at the top, but we should say here that we deviate a bit in such-and-such way(s).